pkg/hostpath/hostpath.go

```diff
@@ -166,29 +168,51 @@ func (hp *hostPath) getVolumePath(volID string, volCtx map[string]string) string
 	return filepath.Join(hp.root, volID, podNamespace, podName, podUID, podSA)
 }
 
-func createFile(path string, buf []byte) {
-	file, err := os.Create(path)
-	if err != nil {
-		klog.Errorf("error creating file %s: %s", path, err.Error())
-		return
+func commonUpsertRanger(obj runtime.Object, podPath, filter string, key, value interface{}) error {
+	if key != filter {
+		return nil
+	}
+	payload, _ := value.(Payload)
+	podFileDir := filepath.Join(podPath, fmt.Sprintf("%s", key))
+	// So, what to do with error handling.  Errors with filesystem operations
+	// will almost always not be intermittent, but most likely the result of the
+	// host filesystem either being full or compromised in some long running fashion, so tight-loop retry, like we
+	// *could* do here as a result will typically prove fruitless.
+	// Then, the controller relist will result in going through the secrets/configmaps we share, so
+	// again, on the off chance the filesystem error is intermittent, or if an administrator has taken corrective
+	// action, writing the content will be retried.  And note, the relist interval is configurable (default 10 minutes)
+	// if users want more rapid retry...but by default, no tight loop more CPU intensive retry
+	// Lastly, with the understanding that an error log in the pod stdout may be missed, we will also generate a k8s
+	// event to facilitate exposure
+	// TODO: prometheus metrics/alerts may be desired here, though some due diligence on what k8s level metrics/alerts
+	// around host filesystem issues might already exist would be warranted with such an exploration/effort
+	if err := os.MkdirAll(podFileDir, os.ModePerm); err != nil {
+		msg := fmt.Sprintf("unable to make dir %s: %s", podFileDir, err.Error())
```

Comment:
Did you mean to use your new ProcessFileSystemError function here?

Reply:
fixed to use new function @coreydaley-redhat thanks